authorization/v2/uniformResourceIdentifier.go

```diff
@@ -24,6 +24,64 @@ func NewUriFromRequest(resource resources.Resource, handler gin.HandlerFunc, ctx
 	}
 }
 
+// NewURIFromString parses the string representation of a URI into the UniformResourceIdentifier struct.
+// NewURIFromString expects the string to be of the following form, otherwise ErrInvalidURI is returned.
+// hs:<service_name>:<subsystem>:<version>:<category>:<resource_name>?<allowed_arguments>#<permission_metadata>
+func NewURIFromString(source string) (UniformResourceIdentifier, error) {
+	var (
+		arguments map[string]string
+		metadata  map[string]string
+	)
+
+	pathArgumentSplit := strings.Split(source, "?")
+	path := pathArgumentSplit[0]
+
+	if len(pathArgumentSplit) > 2 {
+		return UniformResourceIdentifier{}, ErrInvalidURI
```

Comment:
Could you provide more context in the returned error with error wrapping like I mentioned below?